api/endpoints/user.py

```diff
@@ -16,67 +81,126 @@ def get_users():
         return jsonify({"message": "Must contain username, email and password."})
-    if User.query.filter_by(username=data['username']).first():
-        return jsonify({"message": "The username already exist."})
-    if User.query.filter_by(email=data['email']).first():
-        return jsonify({"message": "The email is already registered."})
-    new_user = User(
-        username=data['username'],
-        email=data['email'],
-        password=data['password']
-    )
+    user = User.query.filter_by(username=data["username"]).first()
+    if user:
+        return jsonify({"message": f"The username {user.username} already exists"})
+    if User.query.filter_by(email=data["email"]).first():
+        return jsonify({"message": "The email already exists"})
+
+    password = generate_password_hash(data["password"])
+    new_user = User(username=data["username"], email=data["email"], password=password)
     db.session.add(new_user)
     db.session.commit()
-    return jsonify({"message": f"The user {data['username']} has been created."})
+    return generate_access_token(new_user.username)
 
 
-@user_endpoint.route('/api/user/update/<int:user_id>', methods=['PUT'])
-def update_user(user_id):
+@user_endpoint.route("/api/user/update", methods=["PUT"])
+@jwt_required
+def update_user():
     """
     Updates username and email of user in the database.
     """
-    data = request.get_json()
-    if "username" not in data or "email" not in data:
-        return jsonify({"message": "Payload must contain username and email."})
-    user_to_update = User.query.get(user_id)
+    data = request.form
+    try:
+        username = data["username"]
+        email = data["email"]
+        password = data["password"]
+        photo = request.files["photo"]
+    except KeyError:
+        return jsonify({"error": "KeyError exception occurred"})
+
+    if "username" not in data or "email" not in data or "password" not in data:
+        return jsonify({"message": "Must contain username, email and password."})
+    elif photo.filename == "":
+        return jsonify({"message": "You need to upload a profile picture"}), 401
+
+    current_user = get_jwt_identity()
+    user_to_update = User.query.filter_by(username=current_user).first()
     if not user_to_update:
-        return jsonify({"message": f"The user with id {user_id} does not exist"})
-    user_to_update.username = data['username']
-    user_to_update.email = data['email']
+        return jsonify(
+            {
+                "message": f"The user {current_user} does not exist or you may need a new token"
+            }
+        )
+    user_to_update.username = username
+    user_to_update.email = email
+    user_to_update.password = generate_password_hash(password)
+    user_to_update.profile_image = save_profile_picture(photo)
     db.session.commit()
-    return jsonify({"message": "The username and email has been updated."})
+    return jsonify({"message": f"The user {username} has been updated."})
```

Comment:
lets return and show the json form of the updated user unlike the 
`The user {username} has been updated.` action message.

Reply:
resolved